tests/blocks/test_core.py

```diff
@@ -767,6 +767,23 @@ def save_block_flow():
         block = Test.load("test")
         assert block.a == "foo"
 
+    async def test_save_protected_block_with_new_block_schema_version(self, session):
+        """
+        This testcase would fail when block protection was enabled for block type
+        updates and block schema creation.
+        """
+        await models.block_registration.run_block_auto_registration(session=session)
+        await session.commit()
+
+        JSON._block_schema_version = "not the prefect version"
```

Comment:
I'd use `uuid4().hex` to clarify that we're testing a case where the version is one that does not exist.

Reply:
👍 Added a comment for clarity as well